spring-jms/src/main/java/org/springframework/jms/listener/AbstractPollingMessageListenerContainer.java

```diff
@@ -248,7 +248,19 @@ protected boolean receiveAndExecute(
 				rollbackOnException(this.transactionManager, status, ex);
 				throw ex;
 			}
-			this.transactionManager.commit(status);
+			try {
+				this.transactionManager.commit(status);
+			}
+			catch (Throwable ex) {
+				if (status.isCompleted()) {
+					if (logger.isDebugEnabled()) {
+						logger.debug("Caught exception committing transaction and rolled back: " + ex);
+					}
+				}
+				else {
+					throw ex;
```

Comment:
Should this be wrapped in an illustrative unchecked exception?  As-is, this shouldn't compile because `Throwable` is checked, right?

Reply:
It looks like it should be a syntax error, this is:
```java
public class TestThrowable {

    public static void main(String[] args) {
        throw new Throwable("Throw me");
    }
}
```

However catching and re-throwing throwable is valid:
```java
public class TestThrowable {

    public static void main(String[] args) {
        try {
            System.out.println("But in a try?");
        } catch (Throwable e) {
            if (true) {
                throw e;
            }
        }
    }
}
```
Outputs:
```
But in a try?
```

Since I don't know what the infrastructure exception is, the only super class is throwable. I don't think I should wrap it in anything in case anything above might handle the original cause and not my wrapper, I was trying to avoid changing anything more behaviour than I had to here.

Reply:
Ahh, right, since `PlatformTransactionManager.commit(..)` doesn't declare any checked exceptions, the compiler knows that (in this case) Throwable must necessarily be an unchecked exception.  If you added something to the try-block that could throw a checked exception, only _then_ `throw e;` would fail to compile.  Forgot about that nicety; thanks for the patience!